src/lib.rs

```diff
@@ -29,15 +29,44 @@ mod dlmalloc;
 #[cfg(all(feature = "global", not(test)))]
 mod global;
 
+/// A platform interface
+pub trait System {
+    /// Allocates a memory region of `size` bytes
+    unsafe fn alloc(size: usize) -> (*mut u8, usize, u32);
+
+    /// Remaps a memory region
+    unsafe fn remap(ptr: *mut u8, oldsize: usize, newsize: usize, can_move: bool) -> *mut u8;
+
+    /// Frees a part of a memory region
+    unsafe fn free_part(ptr: *mut u8, oldsize: usize, newsize: usize) -> bool;
+
+    /// Frees an entire memory region
+    unsafe fn free(ptr: *mut u8, size: usize) -> bool;
+
+    /// Indicates if the platform can release a part of memory
+    fn can_release_part(flags: u32) -> bool;
+
+    /// Acquires the global lock
+    #[cfg(feature = "global")]
+    fn acquire_global_lock();
+
+    /// Releases the global lock
+    #[cfg(feature = "global")]
+    fn release_global_lock();
```

Comment:
This is not a correct use of the feature mechanism. If implementors of `System` compile correctly without the `global` feature enabled, they should continue doing so if the feature is enabled.